### PR TITLE
feature: generation of .gitignore(s) with prompt / direct command

### PR DESCRIPTION
**Pull request by meta-boy**
Added helper functions to implement the .gitignore generation command.

WIP towards resolving #55 (and #56?).

### CODE REVIEW

**Comment by meta-boy** [comment]
> Well, I guess getting the ignore files from github is actually better than saving the gitignore files locally. It's 2020 after all... no need to have offline-fallbacks. But we should def. have some exception handling for offline-users so that the cli doesn't just crash.

Yes the update function just exists as of now, will implement it in a try-catch block while actually writing the command. And about the multiple `.gitignore` files, I believe (correct me if it's wrong) it's the convention to only use one `.gitignore`, if in fact, the user wants to remove the `.gitignore` rules then he can just delete the lines from the file itself.

**Comment by ZenaMel** [comment]
After reading through this [thread](https://stackoverflow.com/questions/3305869/are-multiple-gitignores-frowned-on) I'd lean more towards a single gitignore file now. Of course we can add comment blocks around the different sections of the gitignore so that users can quickly delete whqt they want. I haven't checked but I think the official gitignores have comment blocks already anyway.

**Comment by meta-boy** [comment]
I have not yet added the automatic gitignore repo update function to the code,  as I was not sure if to add it. Let me know if it's okay to add it.

File: code-base/src/main.py [hunk 1]
```diff
@@ -1,75 +1,24 @@
...
-    if project_type.lower() in project_types:
-        return True
-    return False
-
-# Code execution
-# ------------------------------------------------------------------
-main()
+@click.argument('project_type')
+@click.option('--name', '-n', help="Name of your project")
+@click.option('--gitignore', '-g', help="Space separated names of langauge / framework (s) you want to add gitignore rules for. " +
```
Comment (ZenaMel) [inline code-base/src/main.py:9]: framework (s) --> framework(s)

File: code-base/src/main.py [hunk 2]
```diff
@@ -1,75 +1,24 @@
...
+              "Example: -g \"python java node\"", default='')
+def main(project_type: str, name: str, gitignore: str):
+
+    if not check_project_type(project_type):
+        return click.echo('Invalid Project type.\nPossible project types are ' +
+                          '-> ' + ', '.join(project_types))
```
Comment (ZenaMel) [inline code-base/src/main.py:16]: nice

File: code-base/src/utils/misc.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    else:
+        gitignores = gitignore_string.split(' ')
+
+    click.echo("Creating Project...")
+    if basic_project(project_name, gitignores):
+        click.echo("Finished Creating Project")
+        click.echo("[Summary of Project]")
+        click.echo("Project Name: " + project_name)
+        click.echo("Project Type: Web-app")
+        click.echo("Project Location: " + os.getcwd())
```
Comment (ZenaMel) [inline code-base/src/utils/misc.py:66]: also, nice.

**Review by ZenaMel — APPROVED** [review]
lgtm

**Pull request merged**